cmd/config-current.go

```diff
@@ -287,6 +287,7 @@ func validateConfig(s config.Config) error {
 	if _, err := logger.LookupConfig(s); err != nil {
 		return err
 	}
+	env.SetEnvOn()
```

Comment:
defer env.SetEnvOff() - is also needed @poornas

Reply:
At the start of validateConfig() there is a `defer env.SetEnvOn()` - leaving env ON seems to be the default state, so there would be no need to set the env off. Further, this is the final check before returning from validateConfig() so it would not interfere with any other config validation.

Reply:
If you see the code it is explicitly mentioned that we turn-off such that config is set/unset not based on ENV only the overrides are based on ENV.

We only do EnvOn for KMS because KMS has an env only value which needs validation. Because if you do this a config setting cannot be set if an ENV is enabled. That was never the intention - config should be allowed to be set/unset regardless of what ENV is set.

I don't think this PR is fixing the issue #9066

Reply:
globalNotificationSys.ConfiguredTargetIDs() returns all targets for which event notifications are currently configured - notify.TestNotificationTargets() validates the incoming config against the list of existing notification targets configured for bucket events. It is not possible to validate whether incoming config is deleting an existing event notification target without accounting for notification settings set via env. variables.

Reply:
We should limit the scope here @poornas this PR should fix the NotificationTargets for only things config specific not ENV, if ENV is already set it won't even come here and the server has already started fine.

This code should be concerned only with config and should use/implement code which only concerns itself with configured targets which are config based not ENV. That is the bug that needs to be fixed.

Reply:
@harshavardhana, updated this PR to exclude env set targets from the validation of incoming config